### ISSUE TITLE
Output for Snom-Phones: Order of numbers and empty entries are not the way Snom wants them

### ISSUE THREAD

**Issue by shamrockDE** [Massi-X/freepbx-phonemiddleware#11]
**Is your feature request related to a problem? Please describe.**
I've installed version 1.3 today and i had again a closer look to the xml-output. I have to change two points manually, to be able to load the XML-File to my Snom-Dect-base.

**Describe the solution you'd like**
**1.:** The order of the numbers for each name has to be **exactly**
- telephone
- office
- mobile

otherwise some numbers will not be loaded.
With version 1.3 (and before) the order of the numbers depends on the content (empty or not).

**2.:** Empty numbers are written with syntax "`<telephone/>`", which is correct html-syntax. But Snom expects empty fields written with syntax "`<telephone></telephone>`", otherwise i get an error during loading.

**Describe alternatives you've considered**
I made both changes manually in the xml-file, then i could load it without problems.

**Additional context**

Maybe this example from the users manual will help. It has more details, than the example on the homepage from snom:
```
<IPPhoneDirectory>
     <DirectoryEntry>
          <Name>Name1</Name>
          <Telephone>phone number1</Telephone>
          <Office>office phone number1</Office>
          <Mobile>mobile phone number1</Mobile>
     </DirectoryEntry>
     <DirectoryEntry>
          <Name>Name2</Name>
          <Telephone></Telephone>
          <Office>office phone number2</Office>
          <Mobile></Mobile>
     </DirectoryEntry>
     <DirectoryEntry>
          <Name>Name3</Name>
          <Telephone>phone number3</Telephone>
          <Office> office phone number3 </Office>
          <Mobile></Mobile> 
     </DirectoryEntry>
     <DirectoryEntry>
          <Name>Name4</Name>
          <Telephone>phone number4</Telephone> 
          <Office></Office>
          <Mobile>mobile phone number4</Mobile>
    </DirectoryEntry>
     <DirectoryEntry>
          <Name>Name5</Name>
          <Telephone>phone number5</Telephone> 
          <Office></Office>
          <Mobile> </Mobile>
    </DirectoryEntry>
</IPPhoneDirectory>
```

**Comment by Massi-X** [comment]
Wow snom are very tough! I will include a fix with the next version
Thank you as always for the test and bug report!

**Comment by Massi-X** [comment]
@shamrockDE I checked the empty tags thing, but the current output is exactly what you ask! Indeed the library outputs for example `<Mobile></Mobile>` and NOT `<Mobile/>`.
It's not possible that the output was like you showed. For the order I will implement it as I said before!

**Comment by Massi-X** [comment]
And by curiosity, the phone doesn't complain about missing tags between other filled tags? From an example you made:
```
<DirectoryEntry>
  <Name>Name2</Name>
  <Telephone></Telephone>
  <Office>office phone number2</Office>
  <Mobile></Mobile>
 </DirectoryEntry>
```

This works fine even if the `Telephone` tag is empty, followed by the filled `Mobile` tag?

**Comment by Massi-X** [comment]
Fixed with version 2.0.0. Downloads moved [here](https://phonemiddleware.000webhostapp.com/releases/)
